Release/src/http/client/http_client_asio.cpp

```diff
@@ -1920,15 +1915,34 @@ class asio_context final : public request_context, public std::enable_shared_fro
         enum timer_state
         {
             created,
-            started,
+            connecting,
+            communicating,
             stopped,
             timedout
         };
 
+        void schedule_timeout()
+        {
+            const auto duration = (m_state == connecting)
+                                    ? m_connection_timeout_duration
+                                    : m_timeout_duration;
+
+            const auto cancelled_handlers = m_timer.expires_from_now(duration);
+            if (cancelled_handlers > 0 || m_state == connecting) {
```

Comment:
Note to self: This might actually fail to register a timeout handler if the connection was reused. In that case, the initial registration would happen with `m_state == communicating`.